dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -250,13 +256,34 @@ private String readFileFromRepo(String fileName, String reference, GHRepository
                 start.add(folders.get(i));
                 String partialPath = Joiner.on("/").join(start);
                 try {
-                    Pair<GHContent, String> innerContent = getContentAndMetadataForFileName(partialPath, reference, repo);
-                    if (innerContent != null && innerContent.getLeft().getType().equals("symlink")) {
-                        // restart the loop to look for symbolic links pointed to by symbolic links
-                        List<String> newfolders = Lists.newArrayList(innerContent.getRight().split("/"));
-                        List<String> sublist = folders.subList(i + 1, folders.size());
-                        newfolders.addAll(sublist);
-                        folders = newfolders;
+                    Pair<GHContent, String> innerContent = getContentAndMetadataForFileName(partialPath, reference, repo, submoduleRedirected);
+                    if (innerContent != null) {
+                        if (innerContent.getLeft().getType().equals("symlink")) {
+                            // restart the loop to look for symbolic links pointed to by symbolic links
+                            List<String> newfolders = Lists.newArrayList(innerContent.getRight().split("/"));
+                            List<String> sublist = folders.subList(i + 1, folders.size());
+                            newfolders.addAll(sublist);
+                            folders = newfolders;
+                        } else if (innerContent.getLeft().getType().equals("submodule")) {
+                            String otherRepo = innerContent.getLeft().getGitUrl();
+                            if (otherRepo == null) {
+                                // likely means this submodule is not on GitHub, rest API reports it as null
```

Comment:
Interesting, that's possible? Hmm, I guess you push one repo but not the other?

Reply:
Yeah, one of my test cases is a bitbucket submodule in a github repo. 
I think git (the local tool) is smart enough to use two different sets of credentials